### PR TITLE
Adds ONNX Mul op

### PR DESCRIPTION
**Pull request by Guriido**
Adds Mul operator support

### CODE REVIEW

File: nnoir-onnx/nnoir_onnx/operators/mul.py [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+
+    def __init__(self, node, *args):
+        super().__init__(node, *args)
+
+    def to_function(self, env, constants):
+        [a, b] = self.node.input
+        if a in constants and b not in constants:
+            return [MulConstant([b], list(self.node.output), value=encode_ndarray(constants[a]))]
+        elif a not in constants and b in constants:
+            return [MulConstant([a], list(self.node.output), value=encode_ndarray(constants[b]))]
```
Comment (notogawa) [inline nnoir-onnx/nnoir_onnx/operators/mul.py]: MulConstant is operation for (dynamic)tensor * (constant)scalar. Scale operation is suitable for this (dynamic)tensor * (constant)tensor case. (This distinction is from historical one)

**Comment by Guriido** [comment]
should I add test case for each combination in {constant, not constant}² for inputs ?

**Comment by notogawa** [comment]
constant x constant case is removed by constant propagation.

**Comment by Guriido** [comment]
It seems that version previous than opset 7 for Mul cannot be run on the latest onnxruntime.
https://github.com/Microsoft/onnxruntime/blob/master/docs/Versioning.md#backwards-compatibility

File: nnoir-onnx/nnoir_onnx/operators/mul.py [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+            axis = env[v].ndim - env[w].ndim
+            if axis == 0 and not unidirectional_broadcastable(env[v].shape, env[w].shape):
+                internal_node = gen_unregisterd_node_name(env)
+                register_node(env, internal_node, env[w])
+                return [
+                    Constant([], [internal_node], value=constants[w]),
+                    Mul([v, internal_node], list(self.node.output))
+                ]
+            else:
+                return [Scale([v], list(self.node.output), axis=axis, W=constants[w], b=None)]
```
Comment (notogawa) [inline nnoir-onnx/nnoir_onnx/operators/mul.py:25]: We can convert all to `Constant+Mul` without check `unidirectional_broadcastable`.

File: nnoir-onnx/test/test_mul.py [hunk 3]
```diff
@@ -0,0 +1,96 @@
...
+from onnx.numpy_helper import from_array
+import onnx
+import numpy as np
+
+from nnoir_onnx.operators.utils import UnsupportedONNXOperation
+from nose.tools import raises
+
+info = make_tensor_value_info
+
+
```
Comment (notogawa) [inline nnoir-onnx/test/test_mul.py:13]: Plz, add testcase `len(a_shape) != len(b_shape)` .

File: nnoir-onnx/nnoir_onnx/operators/mul.py (hunk 2 above)
Comment (Guriido) [reply nnoir-onnx/nnoir_onnx/operators/mul.py:25]: Roger that!

**Pull request merged**